include/picongpu/particles/collision/binary/RelativisticBinaryCollision.hpp

```diff
@@ -0,0 +1,513 @@
+                            float3_COLL const labMomentum1
+                                = precisionCast<float_COLL>(par1[momentum_]) / normalizedWeight1;
+                            float_COLL const mass0 = precisionCast<float_COLL>(picongpu::traits::attribute::getMass(
+                                particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                                par0));
+                            float_COLL const mass1 = precisionCast<float_COLL>(picongpu::traits::attribute::getMass(
+                                particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                                par1));
+                            float_COLL const charge0
+                                = precisionCast<float_COLL>(picongpu::traits::attribute::getCharge(
+                                    particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                                    par0));
+                            float_COLL const charge1
+                                = precisionCast<float_COLL>(picongpu::traits::attribute::getCharge(
+                                    particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                                    par1));
+                            float_COLL const gamma0 = picongpu::gamma<float_COLL>(labMomentum0, mass0);
+                            float_COLL const gamma1 = picongpu::gamma<float_COLL>(labMomentum1, mass1);
+
+                            //                            PMACC_DEVICE_ASSERT_MSG(
+                            //                                math::sqrt(pmacc::math::abs2(labMomentum0)) / gamma0 /
+                            //                                mass0 < c, "labVelocity0: %e",
+                            //                                sqrt(pmacc::math::abs2(labMomentum0)) / gamma0 / mass0);
+                            //                            PMACC_DEVICE_ASSERT_MSG(
+                            //                                math::sqrt(pmacc::math::abs2(labMomentum1)) / gamma1 /
+                            //                                mass1 < c, "labVelocity1: %e",
+                            //                                sqrt(pmacc::math::abs2(labMomentum1)) / gamma1 / mass1);
+                            // [Perez 2012] (1)
+                            // todo:: maybe this also with doubles?
+                            float3_COLL const comsVelocity
+                                = (labMomentum0 + labMomentum1) / (mass0 * gamma0 + mass1 * gamma1);
+                            float_COLL const comsVelocityAbs2 = pmacc::math::abs2(comsVelocity);
+                            float3_COLL comsMomentum0;
+                            float_COLL gammaComs, factorA, coeff0, coeff1;
+
+                            if(comsVelocityAbs2 != 0.0_COLL)
+                            {
+                                // written as (1-v)(1+v) rather than (1-v^2) for better performance when v close to c
```

Comment:
Is this a todo?

Reply:
I moved this one line down. Not really, it's already like that.

Reply:
I think it's to avoid potential accuracy issues, not for better performance.

Reply:
okay